server/metrics/metrics.go

```diff
@@ -28,23 +28,40 @@ func SinceInMicroseconds(start time.Time) float64 {
 	return float64(time.Since(start).Microseconds())
 }
 
+// SinceInSeconds gets the time since specified start in seconds.
+func SinceInSeconds(start time.Time) float64 {
+	return float64(time.Since(start).Seconds())
+}
+
 // Metrics is the main structure for starting the metrics endpoints.
 type Metrics struct {
 	config                        *libconfig.MetricsConfig
-	metricOperations              *prometheus.CounterVec
-	metricOperationsLatency       *prometheus.GaugeVec
-	metricOperationsLatencyTotal  *prometheus.SummaryVec
-	metricOperationsErrors        *prometheus.CounterVec
-	metricImagePullsByDigest      *prometheus.CounterVec
-	metricImagePullsByName        *prometheus.CounterVec
-	metricImagePullsByNameSkipped *prometheus.CounterVec
-	metricImagePullsFailures      *prometheus.CounterVec
-	metricImagePullsSuccesses     *prometheus.CounterVec
+	metricOperations              *prometheus.CounterVec // to be deprecated
+	metricOperationsLatency       *prometheus.GaugeVec   // to be deprecated
+	metricOperationsLatencyTotal  *prometheus.SummaryVec // to be deprecated
+	metricOperationsErrors        *prometheus.CounterVec // to be deprecated
+	metricImagePullsByDigest      *prometheus.CounterVec // to be deprecated
+	metricImagePullsByName        *prometheus.CounterVec // to be deprecated
+	metricImagePullsByNameSkipped *prometheus.CounterVec // to be deprecated
+	metricImagePullsFailures      *prometheus.CounterVec // to be deprecated
+	metricImagePullsSuccesses     *prometheus.CounterVec // to be deprecated
 	metricImagePullsLayerSize     prometheus.Histogram
-	metricImageLayerReuse         *prometheus.CounterVec
+	metricImageLayerReuse         *prometheus.CounterVec // to be deprecated
 	metricContainersOOMTotal      prometheus.Counter
-	metricContainersOOM           *prometheus.CounterVec
+	metricContainersOOM           *prometheus.CounterVec // to be deprecated
 	metricProcessesDefunct        prometheus.GaugeFunc
+
+	metricOperationsTotal                   *prometheus.CounterVec
+	metricOperationsLatencySeconds          *prometheus.GaugeVec
+	metricOperationsLatencySecondsTotal     *prometheus.SummaryVec
```

Comment:
I think these three metrics produce a lot of duplicate information. A Prometheus summary will include a total count and a sum. So at least `metricOperationsTotal` is not needed.
I wonder how useful `metricOperationsLatencySeconds` is as a gauge. A gauge would typically be used to get a moment in time value for something that can go up or down. If it is more interesting to look at changes over a time period a counter and `rate()` is often a good alternative. See also https://prometheus.io/docs/practices/instrumentation/#counter-vs-gauge-summary-vs-histogram.

So unless you really need the gauge I'd say that the summary `metricOperationsLatencySecondsTotal` has everything you need. The summary metric should then also be renamed to `metricOperationsLatencySeconds` as this type will add the buckets and a count and sum metric with the respective suffixes (see https://prometheus.io/docs/practices/histograms/)

Reply:
As of now we're not dropping any existing metric in order to introduce a non-breaking change. We're just adding new metrics for idiomatic existing ones that follow Prometheus naming best practices and reduce cardinality. However, the insight shared is helpful and probably we can drop some metrics like the ones stated in a future release when refactor our metrics collector again to remove metrics instead.